### PR TITLE
Query optimising

### PR DESCRIPTION
**Pull request by Br3nda**
Reduces the number of queries on the home page, by:

* Moving interesting to a scope, so it's performed in the database, instead of pulling all records and running ruby
* Adding .includes() for several things we need at display time.
* Adding the bullet gem, which displays in-efficient queries in development

Hoping this helps #1250

### CODE REVIEW

File: app/models/planting.rb [hunk 1]
```diff
@@ -8,19 +8,20 @@ class Planting < ActiveRecord::Base
...
+  scope :has_photos, -> { joins(:photos).where("photos.id IS NOT NULL") }
+
   delegate :name,
     :en_wikipedia_url,
     :default_scientific_name,
     :plantings_count,
     to: :crop,
     prefix: true
 
-  default_scope { order("created_at desc") }
```
Comment (Br3nda) [inline app/models/planting.rb]: this was a duplicate (in the file twice)

File: app/models/crop.rb [hunk 2]
```diff
@@ -33,12 +33,15 @@ class Crop < ActiveRecord::Base
...
     # ok on sqlite and psql, but not on mysql
-    where(approval_status: "approved").reorder('random()')
+    approved.reorder('random()')
   }
   scope :pending_approval, -> { where(approval_status: "pending") }
   scope :approved, -> { where(approval_status: "approved") }
   scope :rejected, -> { where(approval_status: "rejected") }
 
+  scope :interesting, -> { approved.has_photos }
+  scope :has_photos, -> { includes(:photos).where.not(photos: { id: nil }) }
```
Comment (pozorvlak) [inline app/models/crop.rb]: Could this go into PhotoCapable?

File: app/models/member.rb [hunk 3]
```diff
@@ -36,6 +36,15 @@ class Member < ActiveRecord::Base
...
   scope :wants_newsletter, -> { where(newsletter: true) }
 
+  scope :interesting, lambda {
+    confirmed
+      .located
+      .recently_signed_in
+      .has_plantings
+  }
+
+  scope :has_plantings, -> { joins(:plantings).where("plantings.id IS NOT NULL") }
```
Comment (pozorvlak) [inline app/models/member.rb]: I slightly prefer the `where.not(plantings: { id: nil }) }` style, but no biggie.

Comment (Br3nda) [inline app/models/member.rb]: oh yes, i didn't know i could do that! looks better

File: app/models/crop.rb (hunk 2 above)
Comment (Br3nda) [inline app/models/crop.rb]: Good idea - i've made a task to do that @ #1294

File: app/models/member.rb (hunk 3 above)
Comment (pozorvlak) [inline app/models/member.rb]: You did it in a couple of other places in this PR :-)

**Pull request merged**